src/components/Add-form.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import { firebase } from '../firebase'
+
+export class AddForm extends Component {
+
+    constructor() {
+        super();
+        this.state = {
+            text: '',
+            textarea: '',
+            plants: []
+        }
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(e) {
```

Comment:
You can replace `handleChange(e) {` with `handleChange = e => {` - the same for handleSubmit - and then you no longer need to bind it in the constructor :)

Reply:
Additionally you can get rid of the constructor and just do 

```js
state = {
  text: '',
  textarea: '',
  plants: []
}
```